generate_all_spectra.py

```diff
@@ -87,98 +87,100 @@ def traverse_and_print_dates(directory,startingday='20230828'):
 
-                    date_cal_lst = [ cal_factor_csv_f.split('/')[-1].split('_')[0]
-                            for cal_factor_csv_f in cal_files]
-                
+                    date_cal_lst = [cal_factor_csv_f.split('/')[-1].split('_')[0]
+                                    for cal_factor_csv_f in cal_files]
+
                     idx_cal = 0
                     for date_cal in date_cal_lst:
                         if date_cal <= str_this_day:
                             cal_fname = date_cal
                             idx_cal += 1
                         else:
                             break
-                    print("Using calibration factor from {}".format(cal_files[idx_cal-1]))
-                    cal_factor_file = cal_files[idx_cal-1]
+                    print("Using calibration factor from {}".format(cal_files[idx_cal - 1]))
+                    cal_factor_file = cal_files[idx_cal - 1]
                 else:
                     cal_factor_file = None
 
                 if cal_strategy == 1:
-                    cal_factor_calfac_x = 1/5e4
-                    cal_factor_calfac_y = 1/5e4
+                    cal_factor_calfac_x = 1 / 5e4
+                    cal_factor_calfac_y = 1 / 5e4
                 else:
                     cal_factor_calfac_x = 1
                     cal_factor_calfac_y = 1
 
-
                 files = glob.glob(full_path + '/*')
                 files.sort()
                 d = dspec.Dspec()
-                
-                d.read(files, source='lwa', timebin=8, freqbin=freq_bin, stokes='IV',freqrange=[29,84],
-                    flux_factor_file=cal_factor_file, 
-                    flux_factor_calfac_x=cal_factor_calfac_x,
-                    flux_factor_calfac_y=cal_factor_calfac_y)
-                
-                d.tofits('/common/lwa/spec_v2/fits/{}{}{}.fits'.format(year,month,day))
-                time_range_all =[ d.time_axis[0] , d.time_axis[-1]]
-                hourly_ranges = divide_time_in_hours(time_range_all[0],time_range_all[1], hour_length=1/24)
+
+                d.read(files, source='lwa', timebin=8, freqbin=freq_bin, stokes='IV', freqrange=[29, 84],
+                       flux_factor_file=cal_factor_file,
+                       flux_factor_calfac_x=cal_factor_calfac_x,
+                       flux_factor_calfac_y=cal_factor_calfac_y)
+
+                d.tofits('/common/lwa/spec_v2/fits/{}{}{}.fits'.format(year, month, day))
+                time_range_all = [d.time_axis[0], d.time_axis[-1]]
+                hourly_ranges = divide_time_in_hours(time_range_all[0], time_range_all[1], hour_length=1 / 24)
 
                 if use_synoptic_spec:
-                    ovsp.plot(datetime(2024, 7, 31), figdir='/common/lwa/spec_v2/daily/',add_logo=add_logo, combine=True)
+                    ovsp.plot(datetime(2024, 7, 31), figdir='/common/lwa/spec_v2/daily/', add_logo=add_logo,
+                              combine=True, clip=[10, 99.5])
```

Comment:
Only this line has been changed. All other modifications are due to automatic code formatting (PEP 8) applied by PyCharm, including removal of redundant imports.